src/main/java/org/opensearch/neuralsearch/processor/normalization/MinMaxScoreNormalizationTechnique.java

```diff
@@ -167,22 +224,197 @@ private float[] getMinScores(final List<CompoundTopDocs> queryTopDocs, final int
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "lower_bounds size %d should be less than or equal to %d",
+                    lowerBoundsParams.size(),
+                    MAX_NUMBER_OF_SUB_QUERIES
+                )
+            );
+        }
+        // parse each lower bound item and put all items in a list
+        List<Pair<LowerBound.Mode, Float>> lowerBounds = lowerBoundsParams.stream().map(this::parseLowerBound).collect(Collectors.toList());
+
+        return Optional.of(lowerBounds);
+    }
+
+    @SuppressWarnings("unchecked")
+    /**
+     * Parse each lower bound item and return a pair of mode and min score
+     * @param boundObj lower bound item provided by the client
+     * @return a single pair of mode and min score
+     */
+    private Pair<LowerBound.Mode, Float> parseLowerBound(Object boundObj) {
+        if ((boundObj instanceof Map) == false) {
+            throw new IllegalArgumentException("each lower bound must be a map");
+        }
+
+        Map<String, Object> lowerBound = (Map<String, Object>) boundObj;
+
+        String lowerBoundModeValue = Objects.toString(lowerBound.get(PARAM_NAME_LOWER_BOUND_MODE), "");
+        LowerBound.Mode mode = LowerBound.Mode.fromString(lowerBoundModeValue);
+        float minScore = extractAndValidateMinScore(lowerBound);
+
+        return ImmutablePair.of(mode, minScore);
+    }
+
+    private float extractAndValidateMinScore(Map<String, Object> lowerBound) {
+        Object minScoreObj = lowerBound.get(PARAM_NAME_LOWER_BOUND_MIN_SCORE);
+        if (minScoreObj == null) {
+            return LowerBound.DEFAULT_LOWER_BOUND_SCORE;
+        }
+        try {
+            float minScore = LowerBound.DEFAULT_LOWER_BOUND_SCORE;
+            if (Objects.nonNull(lowerBound.get(PARAM_NAME_LOWER_BOUND_MIN_SCORE))) {
+                minScore = Float.parseFloat(String.valueOf(lowerBound.get(PARAM_NAME_LOWER_BOUND_MIN_SCORE)));
+            }
+            Validate.isTrue(
+                minScore >= LowerBound.MIN_LOWER_BOUND_SCORE && minScore <= LowerBound.MAX_LOWER_BOUND_SCORE,
+                "min_score must be a valid finite number between %f and %f",
+                LowerBound.MIN_LOWER_BOUND_SCORE,
+                LowerBound.MAX_LOWER_BOUND_SCORE
+            );
+            return minScore;
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("invalid format for min_score: must be a valid float value", e);
+        }
     }
 
     /**
      * Result class to hold min and max scores for each sub query
      */
     @AllArgsConstructor
     @Getter
-    private class MinMaxScores {
+    private static class MinMaxScores {
         float[] minScoresPerSubquery;
         float[] maxScoresPerSubquery;
     }
+
+    /**
+     * Result class to hold lower bound for each sub query
+     */
+    @Getter
+    public static class LowerBound {
```

Comment:
Since this is a public class anyway. We could move it to a separate class file

Reply:
Public scope is too broad for this class, will change it to package private and keep it under MinMaxTechnique main class